### PR TITLE
HIVE-28054: SemanticException for join condition in subquery

### PR DESCRIPTION
**Pull request by soumyakanti3578**
### What changes were proposed in this pull request?

Cleaned up `getPosition` method in `RexNodeExprFactory`

### Why are the changes needed?

`getPosition` method didn't return the position of a column as soon as it found it in one of the RowResolvers. It threw an Exception if it encountered the column again in `outerRR`. Now it returns early.

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

`mvn test -Dtest=TestMiniLlapLocalCliDriver -Dtest.output.overwrite=true -Dqfile=subquery_join.q`

### CODE REVIEW

**Review by henrib — APPROVED** [review]
LGTM 👍 : clean code with comment and understandable result

File: ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java [hunk 1]
```diff
@@ -165,22 +165,23 @@ protected RexNode createColumnRefExpr(ColumnInfo colInfo, List<RowResolver> rowR
...
-          throw new CalciteSemanticException("Could not resolve column name");
-        }
-        cInfoToRet = tmp;
-        position += rr.getPosition(cInfoToRet.getInternalName());
-      } else if (cInfoToRet == null) {
+
+    for (RowResolver rr: rowResolverList) {
+      ColumnInfo tmp = rr.get(colInfo.getTabAlias(), colInfo.getAlias());
+      if (tmp == null) {
+        // if column in not present in the RR, increment position by size of RR
```
Comment (kasakrisz) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java:173]: typo?
is not present

**Review by kasakrisz — APPROVED** [review]
LGTM

File: ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java (hunk 1 above)
Comment (soumyakanti3578) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/type/RexNodeExprFactory.java:173]: Yes! I will fix it! Thanks for catching it :)

**Review by kasakrisz — APPROVED** [review]

**Pull request merged**